src/main/java/com/gmail/goosius/siegewar/playeractions/PlunderTown.java

```diff
@@ -231,4 +281,23 @@ private static void transferPlunderToNation(Siege siege, Nation nation, double t
 			}
 		}
 	}
+	
+	private static Map<Resident, Integer> gatherResidentsShareMap(Siege siege) {
```

Comment:
:construction: 
- I like the idea to modularize this logic!
- This method actually holds the key to fixing the existing bug:
   1. First, rename siege.getAttackerSiegeContributors()  to siege.getResidentTimedPointContributors()
   ........ because now we know that it is not only attacking resident who can contribute, but defending ones too
   2. Also rename the associated variable to residentTimedPointContributors
   3. Now, look into where that variable is written to..........and fix it so that the right residents get added to it i.e.
       - For Conquest/Liberation/Suppression sieges,  add soldiers from the attacking nation (& allies)
       - For Revolt  sieges, add soldiers from  the defending nation (& allies)
   4. When this is done, I believe it will either fix the bug or be very close.